llvm-external-projects/iree-dialects/include/iree-dialects/Transforms/ListenerGreedyPatternRewriteDriver.h

```diff
@@ -4,37 +4,79 @@
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+// This is modified version of GreedyPatternRewriterDriver.h. A RewriteLister
+// is passed to all entry points.
+
 #include "iree-dialects/Transforms/Listener.h"
 #include "mlir/IR/PatternMatch.h"
 #include "mlir/Rewrite/FrozenRewritePatternSet.h"
 #include "mlir/Transforms/GreedyPatternRewriteDriver.h"
 
 // The following are iree-dialects extensions to MLIR.
 namespace mlir {
-struct GreedyRewriteConfig;
 struct RewriteListener;
 
-/// Applies the specified patterns on `op` alone while also trying to fold it,
-/// by selecting the highest benefits patterns in a greedy manner. Returns
-/// success if no more patterns can be matched. `erased` is set to true if `op`
-/// was folded away or erased as a result of becoming dead. Note: This does not
-/// apply any patterns recursively to the regions of `op`. Accepts a listener
-/// so the caller can be notified of rewrite events.
-LogicalResult applyPatternsAndFoldGreedily(
-    Operation *op, const FrozenRewritePatternSet &patterns,
-    const GreedyRewriteConfig &config, RewriteListener *listener);
+/// Rewrite ops in the given region, which must be isolated from above, by
```

Comment:
I *think* @nicolasvasilache removed the isolated-from-above restriction from this version, but it needs to be double-checked.

Reply:
We should be able to use `applyOpPatternsTrackAndFold` overload with an `ArrayRef<Operation *>` instead of a region now.